### PR TITLE
Events and users

### PR DESCRIPTION
**Pull request by angelacuahutle**
Dear TSE, 

I'm doing a pull request as I did my best but my steps are slow. I have done almost all requirements so far but still some points missing so I'm asking for guidance with what I have done so far.

The following points are missing and still working on them. 

- [x] Separate this list of “attended_events” into either events that have occurred in the past or events that will occur in the future. You’ll get some good practice building queries and working with dates. Keep this logic in view and do not put separate method calls in the controller.

Finishing Touches
- [x] Separate the past and upcoming events on the Event’s Index page by creating two class methods on the Event model (e.g. Event.past).
- [x] Refactor the past and upcoming class methods into simple scopes.
- [x] Put navigation links across the top to help you jump around.
- [x] Extra Credit: Allow users to edit and delete the events they created.
- [ ] Extra Credit: Allow users to remove themselves as an attendee to their “attended_events”.
- [ ] Extra Credit: Allow users to invite other users to events and only allow invited users to attend an event.

### CODE REVIEW

**Review by MkrtichSargsyan — COMMENTED** [review]
## _Status:_ Invalid submission warning

## Hello Angela Cuahutle Navarro

This Code Review Request will be marked as invalid in your Dashboard because you didn't follow the correct [Git Flow](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) as required for this project. 

Also, don't forget to add actions on Github

Please make the necessary changes and submit another code review request once you are ready! :slightly_smiling_face:.


> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

**Comment by angelacuahutle** [comment]
- Pointed feature events-and-users branch to development instead of master to follow Git Flow convention
- Linters added and fixed

**Review by MkrtichSargsyan — COMMENTED** [review]
## _Status:_ Invalid submission warning

## Hello Guys


This Code Review Request will be marked as invalid in your Dashboard because you didn't follow the correct [Git Flow](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) as required for this project. 
Kindly make the development branch the default branch


Please make the necessary changes and submit another code review request once you are ready! :slightly_smiling_face:.


> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

**Review by carlos-ssh — COMMENTED** [review]
Hi @angelacuahutle , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that You need to follow git-flow in your project. Kindly check general requirements.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by angelacuahutle** [comment]
Dear TSE, 

I'm doing a pull request as I did my best but my steps are slow. I have done almost all requirements so far but still some points missing so I'm asking for guidance with what I have done so far.

The following points are missing and still working on them. 

- [ ] Separate this list of “attended_events” into either events that have occurred in the past or events that will occur in the future. You’ll get some good practice building queries and working with dates. Keep this logic in view and do not put separate method calls in the controller.

Finishing Touches
- [ ] Separate the past and upcoming events on the Event’s Index page by creating two class methods on the Event model (e.g. Event.past).
- [ ] Refactor the past and upcoming class methods into simple scopes.
- [ ] Put navigation links across the top to help you jump around.
- [ ] Extra Credit: Allow users to edit and delete the events they created.
- [ ] Extra Credit: Allow users to remove themselves as an attendee to their “attended_events”.
- [ ] Extra Credit: Allow users to invite other users to events and only allow invited users to attend an event.

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :red_circle: :stop_sign: 
---------------
Hello @angelacuahutle, great job you are doing on this project. 
TO HIGHLIGHT: 

- Your project is professional 
- Your repo is professional 
- Well structured files  
- Well use of the MVP

However, there are a few changes to make to complete this project :hammer_and_wrench: see them below :eyes: :point_down: 
------------------

- [ ] Event's index page displays past and upcoming events lists. Currently, the index page is showing all the events and not separated in past and upcoming. Please fix this. 

past_upcoming_events_requirement (2)

- [x] The index page does not show the name of the event. This is because the title is not saving. 

Screenshot from 2021-07-25 22-04-15

- [x] The event's show page displays the list of attendants. I cannot see the attendance because I cannot be able to attend an event. Please fix this. 

events_attendees_requirement (1)

- [ ] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming. I cannot access to this because there is no link to the user page. Please add a link. 

user_events_requirement (3)

- [ ] When I try to go to the user's show page I got this error. 

Screenshot from 2021-07-25 22-09-28

On your events model.

- [ ] Scope for upcoming events. 
`scope :upcoming, -> { where('date >= ?', Time.zone.now) }`

- [ ] scope for previous events.
`scope :previous, -> { where('date < ?', Time.zone.now) }`

You have this in the join table you create. Please move them to the events model. 

scopes_requirement

Check your association

- [ ] User model: `has_many :events, foreign_key: :creator_id`

user_assoc_requirement

Migrations

- [ ] Events table has creator_id as the foreign key
```
def change
   create_table :events do |t|
    t.integer :creator_id
  end
end
```
UsersController

- [ ] Lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous`
- [ ] Lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming`

- [ ] On your Readme file Before you Run `rails db:migrate` You need to create the database `rails db:create`. Please add this command. 
- [ ] Your screenshot in the Readme file is broken please fix it. 
- [ ] Please add test to your project. 

unit_tests_requirement (1)
---------------------
Please make these required changes and send it again for another review. :muscle: :100: 
Please reach on SLACK if you need help. 
@ Brian SAmmit Cruz Rodriguez
HAPPY CODING!!!

tenor (1)

**Comment by angelacuahutle** [comment]
Dear TSE, 

I  am working alone so I need more patience as the project has taken a lot of time. 

I am still working in associations and making the final touches. I will appreciate any help as it's taking a lot of my nerve to continue with this project on my own. 

Thanks!!

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+
+  # GET /events or /events.json
```
Comment (carlos-ssh) [inline app/controllers/events_controller.rb]: - Kindly remove unused lines of code and comments from your files.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,65 @@
...
+- Ruby installed
+- At least one web browser
+#### Clone the repository
+- Click on the "code" button above on this page and copy the link to the repository
+- Run `git@github.com:angelacuahutle/private-events.git`
+- Go to the cloned repository run `cd private-events`
+#### Installe required Gem
+- In the terminale, run `bundle install`
+- [optional] Run `bundle update`
+#### Set up the database tables
```
Comment (carlos-ssh) [inline README.md:23]: - Provide additional installations like: webpacker.

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  def new
+    @event = current_user.created_events.new
+  end
+
+  # GET /events/1/edit
+  def edit; end
+
+  # POST /events or /events.json
+  def create
+    @event = current_user.created_events.invitations.build(event_params)
```
Comment (carlos-ssh) [inline app/controllers/events_controller.rb]: - I got an error after tried to create a new post. Kindly work on it.
Screen Shot 2021-08-01 at 10 26 24

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hi @angelacuahutle, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by angelacuahutle** [comment]
Dear TSE, 

After a long time, I have covered all the requirements for this project so far. 

Thanks

**Review by BrianSammit — APPROVED** [review]
STATUS: APPROVED :green_circle: 
-------------------
Hello @angelacuahutle, You did a great job so far and you work really hard on this project :muscle: :+1: 
-----------------------
Now you can move on to the next one 
HAPPY CODING!!!

**Pull request merged**